framework/auth/core.py

```diff
@@ -858,15 +950,76 @@ def get_daily_digest_log_ids(self):
         )
         return self.get_recent_log_ids(since=midnight)
 
-    def merge_user(self, user, save=False):
+    @property
+    def can_be_merged(self):
+        """The ability of the `merge_user` method to fully merge the user"""
+        return (
+            self.is_confirmed is False and
+            self.get_addons() == []
+        )
+
+    def merge_user(self, user):
```

Comment:
The `save` parameter applied to the user being merged into (`self`). We should avoid implicit saves whenever possible.